stellarphot/photometry/photometry.py

```diff
@@ -1,138 +1,421 @@
+            std_pp.append(np.std(annulus_data))
+        photom['sky_per_pix_avg'] = photom['annulus_sum'] / photom['annulus_area']
+        photom['sky_per_pix_med'] = np.array(med_pp) * ccd_image.unit * u.pixel**-1
+        photom['sky_per_pix_std'] = np.array(std_pp) * ccd_image.unit * u.pixel**-1
+
+    # Compute counts using clipped stats on sky per pixel
+    photom['aperture_net_cnts'] = (photom['aperture_sum'].value -
+                                    (photom['aperture_area'].value *
+                                    photom['sky_per_pix_avg'].value))
+    photom['aperture_net_cnts'].unit = ccd_image.unit
+
+    # Fit the FWHM of the sources
+    print(f"{logline} fitting fwhm of all sources ...")
+    fwhm_x, fwhm_y = compute_fwhm(ccd_image, photom, fwhm_estimate=fwhm_estimate, fit=fwhm_by_fit)
+    print("                          ...DONE fitting fwhm of all sources")
+
+    # Deal with bad FWHM values
+    bad_fwhm = (fwhm_x<1) | (fwhm_y<1) # Set bad values to NaN now
+    fwhm_x[bad_fwhm] = np.nan
+    fwhm_y[bad_fwhm] = np.nan
+    photom['fwhm_x'] = fwhm_x * u.pixel
+    photom['fwhm_y'] = fwhm_y * u.pixel
+    photom['width'] = ((fwhm_x + fwhm_y) / 2) * u.pixel
+
+    # Flag sources with bad counts before computing noise
+    # This can happen, for example, when the object is faint
+    # and centroiding is bad.
+    bad_cnts = photom['aperture_net_cnts'].value < 0
+    all_bads = bad_cnts | bad_fwhm
+    photom['aperture_net_cnts'][all_bads] = np.nan
+
+    # Compute and save noise
+    noise = calculate_noise(gain=camera.gain.value,
+                            read_noise=camera.read_noise.value,
+                            dark_current_per_sec=camera.dark_current.value,
+                            counts=photom['aperture_net_cnts'].value,
+                            sky_per_pix=photom['sky_per_pix_avg'].value,
+                            aperture_area=photom['aperture_area'].value,
+                            annulus_area=photom['annulus_area'].value,
+                            exposure=photom['exposure'].value,
+                            include_digitization=include_dig_noise)
+    photom['noise_electrons'] = noise  # Noise in electrons
+    photom['noise_electrons'].unit = u.electron
+    photom['noise_cnts'] = noise / camera.gain.value  # Noise in counts
+    photom['noise_cnts'].unit = ccd_image.unit
+
+    # Compute and save SNR
+    snr = camera.gain.value * photom['aperture_net_cnts'] / noise
+    photom['snr'] = snr
+    photom['mag_error'] = 1.085736205 / snr
 
-    # Obtain flux error and add column to return table
-    noise = np.sqrt(gain * net_flux + n_pix_ap * (1 + (n_pix_ap / n_pix_ann)) *
-                    (gain * (bkgd_pp + N_dark_pp) + (N_R**2)))
-    phot_table['aperture_sum_err'] = noise
+    # Compute instrumental magnitudes
+    photom['mag_inst'] = \
+        (-2.5 * np.log10(camera.gain.value * photom['aperture_net_cnts'].value /
+                            photom['exposure'].value))
 
```

Comment:
I have a mild preference for

```python
photom['mag_inst'] = (
    -2.5 * np.log10(camera.gain.value * photom['aperture_net_cnts'].value /
                            photom['exposure'].value)
)
```

Reply:
Since I have no preferences here, change made.